tests/test_handler.py

```diff
@@ -112,26 +119,70 @@ def test_protected_error() -> None:
 # Python exceptions
 
 
-def test_python_exceptions(res_server_error) -> None:
-
-    # NotImplementedError
+def test_not_implemented_error(res_server_error) -> None:
     response = exception_handler(
         NotImplementedError("This function is not implemented")
     )
+    assert response is not None
     assert response.status_code == status.HTTP_500_INTERNAL_SERVER_ERROR
     assert response.data == res_server_error
 
-    # AttributeError
+
+def test_attribute_error(res_server_error) -> None:
     response = exception_handler(AttributeError())
+    assert response is not None
     assert response.status_code == status.HTTP_500_INTERNAL_SERVER_ERROR
     assert response.data == res_server_error
 
-    # ImportError
+
+def test_import_error(res_server_error) -> None:
     response = exception_handler(ImportError())
+    assert response is not None
     assert response.status_code == status.HTTP_500_INTERNAL_SERVER_ERROR
     assert response.data == res_server_error
 
-    # TypeError
+
+def test_type_error(res_server_error) -> None:
     response = exception_handler(TypeError())
+    assert response is not None
     assert response.status_code == status.HTTP_500_INTERNAL_SERVER_ERROR
     assert response.data == res_server_error
+
+
+# Non-APIException handling disabled in DEBUG mode
+
+
+def test_throttled_exception_debug(settings) -> None:
+    settings.DEBUG = True
+    # Same as normal since APIException instance
+    response = exception_handler(exceptions.Throttled(62))
+    assert response is not None
+    assert response.status_code == status.HTTP_429_TOO_MANY_REQUESTS
+    assert response.data == {
+        "type": "throttled_error",
+        "code": "throttled",
+        "detail": "Request was throttled. Expected available in 62 seconds.",
+        "attr": None,
+    }
+
+
+def test_not_found_exception_debug(settings, res_not_found) -> None:
+    settings.DEBUG = True
+    # Same as normal, since APIException instance
+    response = exception_handler(exceptions.NotFound())
+    assert response is not None
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+    assert response.data == res_not_found
+
+    # Test Django base exception too
+    # Not handled, since not APIException instance
+    # (Django errors only inherit from the Python Exception)
+    response = exception_handler(Http404())
+    assert response is None
```

Comment:
This is tricky. Django errors don't inherit from a common Django base error, just from the Python `Exception`, and checking for each Django error would be cumbersome and brittle. Thoughts? @paolodamico

Reply:
Addressed